lnwallet/channel.go

```diff
@@ -2411,7 +2411,13 @@ func (lc *LightningChannel) createCommitmentTx(c *commitment,
 	// We'll now add all the HTLC outputs to the commitment transaction.
 	// Each output includes an off-chain 2-of-2 covenant clause, so we'll
 	// need the objective local/remote keys for this particular commitment
-	// as well.
+	// as well. For any non-dust HTLCs that are manifested on the commitment
+	// transaction, we'll also record its CLTV which is required to sort the
+	// commitment transaction below. The slice is initially sized to the
+	// number of existing outputs, since any outputs already added are
+	// commitment outputs and should correspond to zero values for the
+	// purposes of sorting.
+	cltvs := make([]uint32, len(commitTx.TxOut))
```

Comment:
I think this needs to first add up to two zero valued CLTV values for the commitment outputs (if present). Otherwise, `len(cltvs) != len(commitTx.TxOut)`. Also since we use append below, this should be: 
```go
cltvs := make([]uint32, 0, len(commitTx.TxOut))
```

Reply:
that's what this does?

Reply:
> The slice is initially sized to the number of existing outputs, since any outputs already added are commitment outputs and should correspond to zero values for the purposes of sorting.